crypto/ed25519.go

```diff
@@ -53,21 +53,21 @@ func (key *Ed25519PrivateKey) FromHex(hexStr string) (err error) {
 }
 
 func (key *Ed25519PrivateKey) FromBytes(bytes []byte) (err error) {
-	if len(bytes) != ed25519.PrivateKeySize {
-		return errors.New("invalid ed25519 private key size")
+	if len(bytes) != ed25519.SeedSize {
+		return fmt.Errorf("invalid ed25519 private key size %d", len(bytes))
```

Comment:
So, it seems that many of the private keys in these libraries use a combination of public and private key, and then it comes up with this "seed" terminology, which is very confusing.

Realistically, we use `PrivateKey` to refer to just this `seed` part normally, as the docs on the ed25519 library says.

Reply:
The previous ed25519 environment I dealt with always stored the 64 byte combo key as the 'private key', and sometimes just the public bytes as the 'public key'. I don't know who made that design decision, it was like that when I got there. As long we're clear and consistent in our docs we're good, but since there's divergence on this across users of ed25519 we should be a little more explicit?

Reply:
It's consistent across the rest of our services and docs, so I'm sticking with it there.

Reply:
Ok. I feel good about having had this discussion. Maybe we don't need another API call (that we don't actually use). Maybe all we need is documentation and better error messages?

Reply:
Yeah, because of the sheer amount of code, I think I'll need to probably clean up most docs and error messages.

Let me address these ones now